internal/car/car.go

```diff
@@ -33,34 +34,37 @@ type Car interface {
 }
 
 type car struct {
-	registry internal.Registry
-	out      io.Writer
-	// patterns just like tar. Ex "car -tf image:tag foo/* bar.txt"
-	patterns                       []string
+	registry     internal.Registry
+	out          io.Writer
+	layerPattern *regexp.Regexp
+	// filePatterns just like tar. Ex "car -tf image:tag foo/* bar.txt"
+	filePatterns                   []string
 	fastRead, verbose, veryVerbose bool
 }
 
 // New creates a new instance of Car
-func New(registry internal.Registry, out io.Writer, patterns []string, fastRead, verbose, veryVerbose bool) Car {
+func New(registry internal.Registry, out io.Writer, layerPattern string, patterns []string, fastRead, verbose, veryVerbose bool) Car {
+	var layerRegexp *regexp.Regexp
+	if layerPattern != "" {
+		layerRegexp = regexp.MustCompile(layerPattern)
```

Comment:
given that `layerPattern` is user-provided, maybe better jus use Compile and return error to users, rather than panic here?

Reply:
sgtm!